Upstream: submitted
Source: https://github.com/the-tcpdump-group/libpcap/pull/488

From 873789e0435f4c840c9232fe3e8e22f29cd0bba2 Mon Sep 17 00:00:00 2001
From: Kylie McClain <somasis@exherbo.org>
Date: Mon, 11 Apr 2016 21:24:04 -0400
Subject: [PATCH] configure.in: Use PKG_PROG_PKG_CONFIG for getting the right
 pkg-config

pkg-config provides a macro that comes with the distribution, and it
provides PKG_PROG_PKG_CONFIG, which should be preferred to using custom
code for detecting it; this can aid in building with cross-compilation,
where target-prefixed pkg-config binaries may be preferred to just
`pkg-config`.

diff -Naur libpcap-1.8.1/configure.ac libpcap-1.8.1.new/configure.ac
--- libpcap-1.8.1/configure.ac	2016-10-26 02:07:59.000000000 +0200
+++ libpcap-1.8.1.new/configure.ac	2016-10-31 17:23:09.899055087 +0100
@@ -1729,45 +1729,44 @@
 fi
 
 if test "x$enable_dbus" != "xno"; then
-	AC_CHECK_PROG([PKGCONFIG], [pkg-config], [pkg-config], [no])
-	if test "x$PKGCONFIG" != "xno"; then
-		AC_MSG_CHECKING([for D-Bus])
-		if "$PKGCONFIG" dbus-1; then
-			AC_MSG_RESULT([yes])
-			DBUS_CFLAGS=`"$PKGCONFIG" --cflags dbus-1`
-			DBUS_LIBS=`"$PKGCONFIG" --libs dbus-1`
-			save_CFLAGS="$CFLAGS"
-			save_LIBS="$LIBS"
-			CFLAGS="$CFLAGS $DBUS_CFLAGS"
-			LIBS="$LIBS $DBUS_LIBS"
-			AC_MSG_CHECKING(whether the D-Bus library defines dbus_connection_read_write)
-			AC_TRY_LINK(
-			    [#include <string.h>
+	AC_MSG_CHECKING([for pkg-config])
+	PKG_PROG_PKG_CONFIG()
+	AC_MSG_CHECKING([for D-Bus])
+	if "$PKG_CONFIG" dbus-1; then
+		AC_MSG_RESULT([yes])
+		DBUS_CFLAGS=`"$PKG_CONFIG" --cflags dbus-1`
+		DBUS_LIBS=`"$PKG_CONFIG" --libs dbus-1`
+		save_CFLAGS="$CFLAGS"
+		save_LIBS="$LIBS"
+		CFLAGS="$CFLAGS $DBUS_CFLAGS"
+		LIBS="$LIBS $DBUS_LIBS"
+		AC_MSG_CHECKING(whether the D-Bus library defines dbus_connection_read_write)
+		AC_TRY_LINK(
+		    [#include <string.h>
 
-			     #include <time.h>
-			     #include <sys/time.h>
+		     #include <time.h>
+		     #include <sys/time.h>
 
-			     #include <dbus/dbus.h>],
-			    [return dbus_connection_read_write(NULL, 0);],
-			    [
+		     #include <dbus/dbus.h>],
+		    [return dbus_connection_read_write(NULL, 0);],
+		    [
 				AC_MSG_RESULT([yes])
-				AC_DEFINE(PCAP_SUPPORT_DBUS, 1, [support D-Bus sniffing])
-				DBUS_SRC=pcap-dbus.c
-				V_INCLS="$V_INCLS $DBUS_CFLAGS"
-			    ],
-			    [
-				AC_MSG_RESULT([no])
-				if test "x$enable_dbus" = "xyes"; then
-				    AC_MSG_ERROR([--enable-dbus was given, but the D-Bus library doesn't define dbus_connection_read_write()])
-				fi
-				LIBS="$save_LIBS"
-			     ])
-			CFLAGS="$save_CFLAGS"
-		else
-			AC_MSG_RESULT([no])
+			AC_DEFINE(PCAP_SUPPORT_DBUS, 1, [support D-Bus sniffing])
+			DBUS_SRC=pcap-dbus.c
+			V_INCLS="$V_INCLS $DBUS_CFLAGS"
+		    ],
+		    [
+ 			AC_MSG_RESULT([no])
 			if test "x$enable_dbus" = "xyes"; then
-				AC_MSG_ERROR([--enable-dbus was given, but the dbus-1 package is not installed])
-			fi
+				AC_MSG_ERROR([--enable-dbus was given, but the D-Bus library doesn't define dbus_connection_read_write()])
+ 			fi
+			LIBS="$save_LIBS"
+		     ])
+		CFLAGS="$save_CFLAGS"
+	else
+		AC_MSG_RESULT([no])
+		if test "x$enable_dbus" = "xyes"; then
+			AC_MSG_ERROR([--enable-dbus was given, but the dbus-1 package is not installed])
 		fi
 	fi
 	AC_SUBST(PCAP_SUPPORT_DBUS)
